bolt/test/AArch64/test-indirect-branch.s

```diff
@@ -0,0 +1,110 @@
+// Test how BOLT handles indirect branch sequence of instructions in
+// AArch64MCPlus builder.
+// This test checks the pattern where there is no shift amount after add
+// instruction. The pattern come from libc, it can be reproduced with
+// a 'static' built binary.
+//
+//   adr     x6, 0x219fb0 <sigall_set+0x88>
+//   add     x6, x6, x14, lsl #2
+//   ldr     w7, [x6]
+//   add     x6, x6, w7, sxtw => no shift amount
+//   br      x6
+//
+// It also tests another case where there is no adrp/add pair.
```

Comment:
nit: replace
```cpp
// It also tests another case where there is no adrp/add pair.
// The pattern also come from libc, and it only represents in the binary
// if the lld linker is used to create the static binary.
// It doesn't occur with GCC ld linker.
```

with something like:
```cpp
// Pattern 2: nop/adr pair is used in place of adrp/add
```

Reply:
Done.

Reply:
Hey @kaadam ,
thanks for the changes, looks good.
Applying the latest suggestion from @yota9 and then we are good to go!